types/src/transaction/authenticator.rs

```diff
@@ -491,6 +520,12 @@ pub enum AccountAuthenticator {
         public_key: secp256k1_ecdsa::PublicKey,
         signature: secp256k1_ecdsa::Signature,
     },
+    SingleKey {
+        authenticator: SingleKeyAuthenticator,
+    },
```

Comment:
Can this be flattened? Like

```rust
SingleKey {
        public_key: AnyKey,
        signature: AnySignature,
},
```

Reply:
It would've been good to flatten it just so we're consistent with the rest of the code.